rotkehlchen/serialization/deserialize.py

```diff
@@ -531,8 +535,16 @@ def deserialize_ethereum_transaction(
     """
     source = 'etherscan' if ethereum is None else 'web3'
     try:
-        tx_hash = deserialize_evm_tx_hash(data['hash'])
         block_number = read_integer(data, 'blockNumber', source)
+
+        if block_number != 0:
+            tx_hash = deserialize_evm_tx_hash(data['hash'])
```

Comment:
Note: you are using different approaches to detect a genesis transaction here and [here](https://github.com/rotki/rotki/pull/4475/files#diff-35d35abdccc263615a58d975992495bf4b2bed7a361a852c121a50e60551db89R357). I would be more consistent and would choose one of them to use everywhere. Or maybe even create an util function to detect whether a transaction is genesis

Reply:
Ah yes, a util function would be better rather than using diff approaches and confusing logic.